R/tags.R

```diff
@@ -383,6 +478,10 @@ throw_if_tag_function <- function(tag) {
 #'   normally be written around this tag. Valid options include `before`,
 #'   `after`, `outside`, `after-begin`, and `before-end`.
 #'   Any number of these options can be specified.
+#' @param .renderHooks A function to call when the `tag` is rendered. This
+#'   function should have at least one argument (the `tag`) and return anything
+#'   that can be converted into tags via [as.tags()]. Multiple hooks may also be
+#'   added to a particular `tag` via [tagAddRenderHook()].
```

Comment:
Winston and I were thinking this would just allow a function, not a list of functions (if you really needed that you could write a function to call the list of functions).

Reply:
https://github.com/rstudio/htmltools/pull/215#discussion_r617005722

Winston asked for a list of functions.

Reply:
Yea, we'll always attach a list of functions to the tag object, but when receiving input it should be a singular `.renderHook`

Reply:
Just to clarify: the data structure for storage would be a list of functions, but this function would accept just a single function, for simplicity and consistency. I think it would be very rare for someone to want to pass in multiple render hook functions; if they really want to do that, they could call `tagAddRenderHook()` (possibly multiple times), or they could wrap their set of functions in their own closure which invokes them.